api/db-cdr/generate-cdr/cloudsql-import.sh

```diff
@@ -34,50 +44,133 @@ then
+      --account $SERVICE_ACCOUNT $INSTANCE | grep serviceAccountEmailAddress \
+      | cut -d: -f2)
+  # Trim leading whitespace from sql service account
+  SQL_SERVICE_ACCOUNT=${SQL_SERVICE_ACCOUNT//[[:blank:]]/}
+
+  echo "Granting GCS access to ${SQL_SERVICE_ACCOUNT} to $bucket_path"
+  # Note, this only applies to files already existing in the bucket.
+  gsutil acl ch -u ${SQL_SERVICE_ACCOUNT}:R $bucket_path
+}
+
+
+
+# Create db from sql file if specified
+create_gs_file=gs://
+if [ "${CREATE_DB_SQL_FILE}" ]
+then
+    create_gs_file=$create_gs_file$BUCKET/$CREATE_DB_SQL_FILE
+    grant_access_to_files $create_gs_file
+    echo "Creating cloudsql DB from sql file ${CREATE_DB_SQL_FILE}"
+    gcloud sql import sql $INSTANCE $create_gs_file --project $PROJECT --database=$DATABASE \
+        --account $SERVICE_ACCOUNT --quiet --async
+    import_wait $create_gs_file 15
+    # If above fails let it die as user obviously intended to create the db
+    gsutil mv $create_gs_file gs://$BUCKET/imported_to_cloudsql/
+fi
+
+#Import sql and csv files, do sql first as they are intended for schema changes and such
+# If file argument passed we just import that one
+sqls=()
+csvs=()
+
+if [ "${FILE}" ]
+then
+  # Just grant access to this file
+  gs_file=gs://$BUCKET/$FILE
+  grant_access_to_files $gs_file
+  echo "Acces granted to $gs_file"
+  if [[ $FILE =~ \.sql.* ]]
+  then
+    sqls[0]=$gs_file
+  elif [[ $FILE =~ \.csv.* ]]
   then
-     echo "Import is still running after ${minutes_waited} minutes."
+    csvs[0]=$gs_file
   else
-    echo "Import finished after ${minutes_waited} minutes."
-    break
+    echo "Error. Cannot import file, $FILE,  that is not a .sql or .csv"
+    exit 1;
   fi
+else
+    grant_access_to_files gs://$BUCKET/*
+    # gsutil returns error if no files match thus the "2> /dev/null || true" part
+    sqls=( $(gsutil ls gs://$BUCKET/*.sql* 2> /dev/null || true))
+    csvs=( $(gsutil ls gs://$BUCKET/*.csv* 2> /dev/null || true))
+fi
+
+for gs_file in "${sqls[@]}"
+do
+    if [ "$gs_file" = "$create_gs_file" ]
+    then
+        echo "Skipping create db sql file ";
+    else
+        gcloud sql import sql $INSTANCE $gs_file --project $PROJECT --account $SERVICE_ACCOUNT --quiet --async
+        import_wait $gs_file 15
+        # Move file to imported dir
+        gsutil mv $gs_file gs://$BUCKET/imported_to_cloudsql/
+    fi
 done
+
+for gs_file in "${csvs[@]}"
+do
+   # Get table name from file
+   filename="${gs_file##*/}"  # gets everything after last /
+   table=${filename%%.*}      # truncates everything starting with first .
+   if [[ $table ]]
+   then
+        echo "Importing file into $table. If table does not exist in the database, this file is moved to imported."
+        gcloud sql import csv $INSTANCE $gs_file --project $PROJECT --quiet --account $SERVICE_ACCOUNT \
+        --database=$DATABASE --table=$table --async
+        import_wait $gs_file 15
```

Comment:
I would have thought large CSV files would take much longer than 15 seconds to import?

Reply:
Right, and I was saying: once you poll every second (which I thought you were doing), it's fine to set a bigger timeout... under normal circumstances you won't hit it.

Anyway, this is fine for now, but if we start noticing occasional failures due to slow imports, we could consider bumping these up.